hack/verify-govet.sh

```diff
@@ -3,82 +3,49 @@
 set -o nounset
 set -o pipefail
 
-GO_VERSION=($(go version))
-
-if [[ -z $(echo "${GO_VERSION[2]}" | grep -E 'go1.[6]') && -z "${FORCE_VERIFY-}" ]]; then
-  echo "Unknown go version '${GO_VERSION}', skipping go vet."
-  exit 0
-fi
-
 OS_ROOT=$(dirname "${BASH_SOURCE}")/..
 source "${OS_ROOT}/hack/lib/init.sh"
 
+os::golang::verify_go_version
+
 cd "${OS_ROOT}"
 mkdir -p _output/govet
 
 os::build::setup_env
 
-FAILURE=false
-test_dirs=$(find_files | cut -d '/' -f 1-2 | sort -u)
-for test_dir in $test_dirs
-do
-  go tool vet -shadow=false $test_dir
-  if [ "$?" -ne 0 ]
-  then
-    FAILURE=true
-  fi
-done
-
-# For the sake of slowly white-listing `shadow` checks, we need to keep track of which
-# directories we're searching through. The following are all of the directories we care about:
-# all top-level directories except for 'pkg', and all second-level subdirectories of 'pkg'.
-ALL_DIRS=$(find_files | grep -Eo "\./([^/]+|pkg/[^/]+)" | sort -u)
+vetexceptions=(
+	"pkg/auth/ldaputil/client.go:[[:digit:]]+: assignment copies lock value to c: crypto/tls.Config contains sync.Once contains sync.Mutex"
+)
 
-DIR_BLACKLIST='./hack
-./pkg/api
-./pkg/authorization
-./pkg/build
-./pkg/client
-./pkg/cmd
-./pkg/deploy
-./pkg/diagnostics
-./pkg/dockerregistry
-./pkg/generate
-./pkg/gitserver
-./pkg/image
-./pkg/oauth
-./pkg/project
-./pkg/quota
-./pkg/router
-./pkg/sdn
-./pkg/security
-./pkg/serviceaccounts
-./pkg/template
-./pkg/user
-./pkg/util
-./test
-./third_party
-./tools'
+function is_vetexception() {
+	text=$1
+	for ve in "${vetexceptions[@]}"; do
+		if [[ -n $(echo "${text}" | grep -Eox "${ve}") ]]; then
+			return 0
+		fi
+	done
+	return 1
+}
 
-for test_dir in $ALL_DIRS
-do
-  # use `grep` failure to determine that a directory is not in the blacklist
-  if ! echo "${DIR_BLACKLIST}" | grep -q "${test_dir}"; then
-    go tool vet -shadow -shadowstrict $test_dir
-    if [ "$?" -ne "0" ]
-    then
-      FAILURE=true
-    fi
-  fi
+FAILURE=false
+test_dirs=$(find_files | cut -d '/' -f 1-2 | sort -u)
+for test_dir in $test_dirs; do
```

Comment:
Since we're in here lets do some Bash cleanup of my legacy code :)

Bracket the var:
```sh
for test_dir in ${test_dirs}; do
                 ^         ^
```